src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxNetworkInterface.cs

```diff
@@ -68,15 +68,35 @@ private static LinuxNetworkInterface GetOrCreate(Dictionary<string, LinuxNetwork
             return lni;
         }
 
-        public override bool SupportsMulticast { get { return _supportsMulticast; } }
+        public override bool SupportsMulticast
+        {
+            get
+            {
+                if (_supportsMulticast.HasValue)
+                {
+                    return _speed.Value;
+                }
+                else
+                {
+                    throw new PlatformNotSupportedException(SR.net_InformationUnavailableOnPlatform);
```

Comment:
I'm not sure. I would imagine problems would arise when application code took a path that was only valid for non-multicast interfaces, and we claimed the interface didn't support multicast when it really did. I don't know if any such code exists or whether it even makes sense for something like that to be written. I think almost all uses of this library are for diagnostic / discovery purposes.